pallets/file-system/src/tests.rs

```diff
@@ -7393,6 +7242,149 @@ mod delete_file_and_pending_deletions_tests {
+					owner_signed.clone(),
+					bucket_id,
+					file_key,
+					location,
+					size,
+					fingerprint,
+					None,
+				));
+
+                // Assert that the correct event was deposited
+                System::assert_last_event(
+                    Event::FileDeletionRequest {
+                        user: owner_account_id.clone(),
+                        file_key,
+                        bucket_id,
+                        msp_id: Some(msp_id),
+                        proof_of_inclusion: false,
+                    }
+                        .into(),
+                );
+
+                // Assert that the pending file deletion request was added to storage
+                assert_eq!(
+                    file_system::PendingFileDeletionRequests::<Test>::get(owner_account_id.clone()),
+                    BoundedVec::<_, <Test as file_system::Config>::MaxUserPendingDeletionRequests>::try_from(
+                        vec![(file_key, bucket_id)]
+                    )
+                        .unwrap()
+                );
+
+                let pending_file_deletion_request_ttl: u32 =
+                    PendingFileDeletionRequestTtl::<Test>::get();
+                let pending_file_deletion_request_ttl: BlockNumberFor<Test> =
+                    pending_file_deletion_request_ttl.into();
+                let expiration_block = pending_file_deletion_request_ttl + System::block_number();
+
+                // Assert that the pending file deletion request was added to storage
+                assert_eq!(
+                    file_system::FileDeletionRequestExpirations::<Test>::get(expiration_block),
+                    vec![FileDeletionRequestExpirationItem {
+                        user: owner_account_id.clone(),
+                        file_key,
+                        bucket_id,
+                        file_size: size,
+                    }]
+                );
+
+                pallet_payment_streams::UsersWithoutFunds::<Test>::insert(owner_account_id.clone(), System::block_number());
+
+                // Roll past the expiration block
+                roll_to(pending_file_deletion_request_ttl + 1);
+
+                // Item expiration should be removed
+                assert_eq!(
+                    file_system::FileDeletionRequestExpirations::<Test>::get(expiration_block),
+                    vec![]
+                );
+
+                // Assert that the pending file deletion request was removed from storage
+                assert_eq!(
+                    file_system::PendingFileDeletionRequests::<Test>::get(owner_account_id.clone()),
+                    BoundedVec::<_, <Test as file_system::Config>::MaxUserPendingDeletionRequests>::default()
+                );
+
+                // Assert that the payment stream was correctly deleted since the user is without funds
+                assert!(pallet_payment_streams::FixedRatePaymentStreams::<Test>::get(msp_id, owner_account_id.clone()).is_none());
```

Comment:
how did the payment stream get automatically deleted?

Reply:
The idea is that when a provider tries to charge a user that is deemed as without funds, the `manage_user_without_funds` logic gets executed, which basically charges the debt of the payment stream from the user deposit and then clears it. After this, the provider is free to delete without penalty any files from the user since the runtime knows that it is not getting paid for them

Reply:
ok I see, it's not part of a separate process instead of having it automatically being deleted when we update the payment stream